crates/ty_python_semantic/resources/mdtest/import/workspaces.md

```diff
@@ -0,0 +1,660 @@
+# Support for Resolving Imports In Workspaces
+
+Python packages have fairly rigid structures that we rely on when resolving imports and merging
+namespace packages or stub packages. These rules go out the window when analyzing some random local
+python file in some random workspace, and so we need to be more tolerant of situations that wouldn't
+fly in a published package, cases where we're not configured as well as we'd like, or cases where
+two projects in a monorepo have conflicting definitions (but we want to analyze both at once).
+
+## Invalid Names
+
+While you can't syntactically refer to a module with an invalid name (i.e. one with a `-`, or that
+has the same name as a keyword) there are plenty of situations where a module with an invalid name
+can be run. For instance `python my-script.py` and `python my-proj/main.py` both work, even though
+we might in the course of analyzing the code compute the module name `my-script` or `my-proj.main`.
+
+Also, a sufficiently motivated programmer can technically use `importlib.import_module` which takes
+strings and does in fact allow syntactically invalid module names.
+
+### Current File Is Invalid Module Name
+
+Relative and absolute imports should resolve fine in a file that isn't a valid module name.
+
+`my-main.py`:
+
+```py
+# TODO: there should be no errors in this file
+
+# error: [unresolved-import]
+from .mod1 import x
+
+# error: [unresolved-import]
+from . import mod2
```

Comment:
Both of these imports fail at runtime with `ImportError: attempted relative import with no known parent package` if you execute `python my-main.py`.

This is _mostly_ unrelated to whether `my-main.py` is a valid or invalid module name.

The only way these two imports can work is if all of these modules are imported as part of the same top-level parent package (that is, the directory containing these files is a Python package, and its parent directory is on the search path). With the name `my-main.py`, the only way that can happen is via `importlib.import_module`.